include/tilck/kernel/term_aux.h

```diff
@@ -58,15 +58,20 @@ term_execute_or_enqueue_action_template(term *t,
 {
    bool was_empty;
 
+   kmutex_lock(&rb_data->lock);
```

Comment:
Grabbing the sleeping lock is inefficient just to write to the ring buffer which is a very fast operation. The overhead of that kind of lock is too high compared to the code is trying to protect. The idea behind the "safe_ringbuf" was it is safe against IRQs as long as on the stack there is only one consumer/reader and all the rest of the IRQs and nested IRQs are producers. See the long comment at the beginning of safe_ringbuf.h. 

It now it turns out that because of user space tests there is a way to violate the expectations in this code, maybe because of kernel preemption, then I guess we should use the basic ringbuf.h implementation and protect it by disabling the interrupts.

Do you mind checking a bit deeper what went wrong?